### PR TITLE
Allows some spec weapons to display ammo counters.

### PR DESCRIPTION
**Pull request by RaymondAllison**
## About The Pull Request

It allows the ZX Shotgun to be used point blankly, and displays most if not all specialist weaponry ammo counts.

## Why It's Good For The Game

Specs can now look at their ammunition count and actually manage it properly, without having to eject the actual magazine to look.

INHERITANCE TOOK CARE OF THE ZX

## Changelog
:cl:
tweak:
/:cl:

### CODE REVIEW

**Comment by RaymondAllison** [comment]
Dont mind the IFF part, had a little trouble on that.

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 1]
```diff
@@ -890,6 +890,7 @@
 	current_mag = /obj/item/ammo_magazine/internal/shotgun/scout
 	gun_skill_category = GUN_SKILL_SPEC
 	aim_slowdown = SLOWDOWN_ADS_SPECIALIST_LIGHT
+	flags_gun_features = GUN_AMMO_COUNTER|GUN_CAN_POINTBLANK
```
Comment (DominikPanic) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]: This is bad, you missed two flags, plus inheritance already takes care of it.

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 2]
```diff
@@ -890,7 +890,7 @@
 	current_mag = /obj/item/ammo_magazine/internal/shotgun/scout
 	gun_skill_category = GUN_SKILL_SPEC
 	aim_slowdown = SLOWDOWN_ADS_SPECIALIST_LIGHT
-	flags_gun_features = GUN_AMMO_COUNTER|GUN_CAN_POINTBLANK
+	flags_gun_features = GUN_AMMO_COUNTER
```
Comment (DominikPanic) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]: Even worse, remove this line entirely.

**Comment by Rohesie** [comment]
Travis failing.

**Comment by spookydonut** [comment]
Changelog in pr description is blank

File: code/modules/projectiles/updated_projectiles/ammo_datums.dm [hunk 3]
```diff
@@ -775,6 +775,7 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
 	name = "supersonic sniper bullet"
 	hud_state = "sniper_supersonic"
 	iff_signal = ACCESS_IFF_PMC
+	flags_ammo_behavior = AMMO_BALLISTIC|AMMO_SKIPS_HUMANS
```
Comment (spookydonut) [inline code/modules/projectiles/updated_projectiles/ammo_datums.dm]: Undocumented change

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 4]
```diff
@@ -35,7 +35,7 @@
                         /obj/item/attachable/lasersight,
                         )
 
-	flags_gun_features = GUN_AUTO_EJECTOR|GUN_WIELDED_FIRING_ONLY
+	flags_gun_features = GUN_AUTO_EJECTOR|GUN_WIELDED_FIRING_ONLY|GUN_AMMO_COUNTER
```
Comment (spookydonut) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]: Undocumented change

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 5]
```diff
@@ -205,7 +205,7 @@
 	force = 17
 	zoomdevicename = "scope"
 	attachable_allowed = list()
-	flags_gun_features = GUN_AUTO_EJECTOR|GUN_WIELDED_FIRING_ONLY
+	flags_gun_features = GUN_AUTO_EJECTOR|GUN_WIELDED_FIRING_ONLY|GUN_AMMO_COUNTER
```
Comment (spookydonut) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]: Undocumented change

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 6]
```diff
@@ -289,7 +289,7 @@
 						/obj/item/attachable/attached_gun/grenade
 						)
 
-	flags_gun_features = GUN_AUTO_EJECTOR|GUN_WIELDED_FIRING_ONLY
+	flags_gun_features = GUN_AUTO_EJECTOR|GUN_WIELDED_FIRING_ONLY|GUN_AMMO_COUNTER
```
Comment (spookydonut) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]: Undocumented change

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 7]
```diff
@@ -862,7 +862,7 @@
 	current_mag = /obj/item/ammo_magazine/rocket/m57a4
 	aim_slowdown = SLOWDOWN_ADS_SUPERWEAPON
 	attachable_allowed = list()
-	flags_gun_features = GUN_WIELDED_FIRING_ONLY
+	flags_gun_features = GUN_WIELDED_FIRING_ONLY|GUN_AMMO_COUNTER
```
Comment (spookydonut) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]: Undocumented change

**Pull request closed without merging**